teuthology/test/test_suite.py

```diff
@@ -66,3 +60,183 @@ def test_null_placeholders_dropped(self):
+        result = suite.get_package_versions(
+            "sha1",
+            "rhel",
+            package_versions=self.pv
+        )
+        expected = deepcopy(self.pv)
+        expected['sha1'].update(dict(rhel="1.1"))
+        assert result == expected
+
+    @patch("teuthology.suite.package_version_for_hash")
+    def test_package_versions_not_found(self, m_package_versions_for_hash):
+        # if gitbuilder returns a status that's not a 200, None is returned
+        m_package_versions_for_hash.return_value = None
+        result = suite.get_package_versions(
+            "sha1",
+            "rhel",
+            package_versions=self.pv
+        )
+        assert result == self.pv
+
+    @patch("teuthology.suite.package_version_for_hash")
+    def test_no_package_versions_kwarg(self, m_package_versions_for_hash):
+        m_package_versions_for_hash.return_value = "1.0"
+        result = suite.get_package_versions(
+            "sha1",
+            "ubuntu",
+        )
+        expected = deepcopy(self.pv)
+        assert result == expected
+
+    def test_distro_has_packages(self):
+        result = suite.has_packages_for_distro(
+            "sha1",
+            "ubuntu",
+            package_versions=self.pv,
+        )
+        assert result
+
+    def test_distro_does_not_have_packages(self):
+        result = suite.has_packages_for_distro(
+            "sha1",
+            "rhel",
+            package_versions=self.pv,
+        )
+        assert not result
+
+    @patch("teuthology.suite.get_package_versions")
+    def test_has_packages_no_package_versions(self, m_get_package_versions):
+        m_get_package_versions.return_value = self.pv
+        result = suite.has_packages_for_distro(
+            "sha1",
+            "rhel",
+        )
+        assert not result
+
+
+class TestDistroDefaults(object):
+
+    def test_distro_defaults_saya(self):
+        assert suite.get_distro_defaults('ubuntu', 'saya') == ('armv7l',
+                                                               'saucy', 'deb')
+
+    def test_distro_defaults_plana(self):
+        assert suite.get_distro_defaults('ubuntu', 'plana') == ('x86_64',
+                                                                'precise',
+                                                                'deb')
+
+    def test_distro_defaults_debian(self):
+        assert suite.get_distro_defaults('debian', 'magna') == ('x86_64',
+                                                                'wheezy',
+                                                                'deb')
+
+    def test_distro_defaults_centos(self):
+        assert suite.get_distro_defaults('centos', 'magna') == ('x86_64',
+                                                                'centos6',
+                                                                'rpm')
+
+    def test_distro_defaults_fedora(self):
+        assert suite.get_distro_defaults('fedora', 'manga') == ('x86_64',
```

Comment:
I'll fix it.  That part of the logic doesn't take machine_type into
consideration at all, which is why it still passes.

On Tue, Jan 13, 2015 at 11:50 AM, Zack Cerza <notifications@github.com>
wrote:

> In teuthology/test/test_suite.py
> <https://github.com/ceph/teuthology/pull/402#discussion-diff-22879743>:
>
> > +        assert suite.get_distro_defaults('ubuntu', 'plana') == ('x86_64',
> > +                                                                'precise',
> > +                                                                'deb')
> > +
> > +    def test_distro_defaults_debian(self):
> > +        assert suite.get_distro_defaults('debian', 'magna') == ('x86_64',
> > +                                                                'wheezy',
> > +                                                                'deb')
> > +
> > +    def test_distro_defaults_centos(self):
> > +        assert suite.get_distro_defaults('centos', 'magna') == ('x86_64',
> > +                                                                'centos6',
> > +                                                                'rpm')
> > +
> > +    def test_distro_defaults_fedora(self):
> > +        assert suite.get_distro_defaults('fedora', 'manga') == ('x86_64',
>
> This is typoed but still passing. A little worrisome
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ceph/teuthology/pull/402/files#r22879743>.
>